app/controllers/api/v1/organizations_controller.rb

```diff
@@ -3,7 +3,43 @@ module V1
     class OrganizationsController < ApiController
       include Api::OrganizationsController
 
+      INDEX_ATTRIBUTES_FOR_SERIALIZATION = %i[
+        id name profile_image slug summary tag_line url
+      ].freeze
+
+      SHOW_ATTRIBUTES_FOR_SERIALIZATION = %i[
+        id username name summary twitter_username github_username url
+        location created_at profile_image slug tech_stack tag_line story
+      ].freeze
+
       before_action :find_organization, only: %i[users listings articles]
+
+      def index
+        per_page = (params[:per_page] || 30).to_i
+        num = [per_page, per_page_max].min
+        page = params[:page] || 1
+
+        organizations = Organization.select(INDEX_ATTRIBUTES_FOR_SERIALIZATION).page(page).per(num)
+
+        render json: organizations
+      end
+
+      # This borrows from the existing "show by username" route to add id lookup as a first step.
+      # The reason is that we want people to access both api/organizations/:username (old style)
+      # as well as api/organizations/:id ("new" style)
+      # Ultimately it might be best to rename the less restful "by username" lookup to a different route.
```

Comment:
I actually don't think we need to manage the data mapping, as ActiveRecord already has technology to map string inputs to integer DB fields:

```ruby
> Organization.where(id: "notremotelyintegerish").to_sql
=> "SELECT \"organizations\".* FROM \"organizations\" WHERE \"organizations\".\"id\" = NULL"
```

I think the issue here is that our system allows organizations (and users and articles and pages and podcast episodes...) to have "integer-like" slugs, which makes this kind of lookup inherently ambiguous. If we have a user with the username `"1337"` and a user with the id `1337`, we can't know which one we mean if we have a flexible endpoint. (Should we raise an exception if we find two? Prefer `id` or prefer `username` match?) We _maybe_ ought to mark "integer-like" slugs and usernames as invalid if we want to be able to lookup objects by _either_ `slug` or `id`. I created [an issue](https://github.com/forem/forem/issues/19758) to discuss it further.

Also, we _can't_ use something like `String#is_an_integer?` to route to "id-based finder", because we (currently) can't assume that an integer-like input is _always_ an id.

For this PR, I think the controller should maybe do something like:

```ruby
finder = Organization.select(ATTRIBUTES_FOR_SERIALIZATION)
@organization = finder.find_by(id: params[:identifier]) || finder.find_by(slug: params[:identifier])
raise ActiveRecord::NotFound unless @organization
```

Reply:
> I think the issue here is that our system allows organizations (and users and articles and pages and podcast episodes...) to have "integer-like" slugs, which makes this kind of lookup inherently ambiguous.
That's a good callout and thanks for creating the issue!

> @organization = finder.find_by(id: params[:identifier]) || finder.find_by(slug: params[:identifier])
+1 on this

Reply:
This was something that Joshua and I had puzzled over a little bit and decided the small chance of overlap between username and id lookups was worth the simplicity of one route to handle both types of parameters. I think you're right that a more generally named parameter makes sense. 

The friction I'm encountering is in the new show route's overlap with the existing username route, I was defaulting to that :username namespace so as not to accidentally disturb how anyone is currently getting data from the API. I was hesitant to rename the route without first figuring out if/how I need to communicate that change.

For the future, I think the best option might be to use a query string parameter to disambiguate. thinking of an edge case like an elite white hat hacker org called "007", they could find themselves via a request like `/api/v1/organizations/007?find_by=username` instead of getting the 7th organization back. 

I'm not sure if that's too big of a change for the scope of this one issue though.

For right now, I love the suggestion to check that the value is itself as an integer. Joshua and I had talked about that too but I forgot to implement it. I'll get that into the action. 

Whatever the more general name is that we use internally, does it make sense to keep the documentation spec as is, separating our implementation from the API interface?

Reply:
Thanks Josh! I had been head down and didn't realize my comment was not sent yet, then hit send before refreshing and seeing yours